fastavro/writer.py

```diff
@@ -239,9 +238,15 @@ def write_record(fo, datum, schema):
     their schema."""
     for field in schema['fields']:
         name = field['name']
+        f_type = field['type']
         if name not in datum and 'default' not in field:
-            raise ValueError('no value and no default for %s' % name)
-        write_data(fo, datum.get(name, field.get('default')), field['type'])
+            if 'null' in f_type:
```

Comment:
I think you can accomplish what you are looking for by just changing the original 
```python
if name not in datum and 'default' not in field:
```

to

```python
if name not in datum and 'default' not in field and 'null' not in field['type']:
```